docs/tutorial.md

```diff
@@ -0,0 +1,514 @@
+
+Again, relations come in handy here as TLS is enabled via relations; i.e. by relating Charmed PostgreSQL to the [TLS Certificates Charm](https://charmhub.io/tls-certificates-operator). The TLS Certificates Charm centralises TLS certificate management in a consistent manner and handles providing, requesting, and renewing TLS certificates.
+
+
+### Configure TLS
+Before enabling TLS on Charmed PostgreSQL we must first deploy the `tls-certificates-operator` charm:
+```shell
+juju deploy tls-certificates-operator --channel=edge --config generate-self-signed-certificates="true" --config ca-common-name="Tutorial CA"
+```
+
+Wait until the `tls-certificates-operator` is up and active, use `juju status --watch 1s` to monitor the progress:
+```
+Model     Controller  Cloud/Region         Version  SLA          Timestamp
+tutorial  overlord    localhost/localhost  2.9.42   unsupported  10:31:40+01:00
+
+App                        Version  Status  Scale  Charm                      Channel  Rev  Exposed  Message
+postgresql                          active      2  postgresql                 edge     281  no       
+tls-certificates-operator           active      1  tls-certificates-operator  edge      22  no       
+
+Unit                          Workload  Agent  Machine  Public address  Ports  Message
+postgresql/0*                 active    idle   0        10.89.49.129           Primary
+postgresql/1                  active    idle   1        10.89.49.197           
+tls-certificates-operator/0*  active    idle   4        10.89.49.185           
+
+Machine  State    Address       Inst id        Series  AZ  Message
+0        started  10.89.49.129  juju-a8a31d-0  jammy       Running
+1        started  10.89.49.197  juju-a8a31d-1  jammy       Running
+4        started  10.89.49.185  juju-a8a31d-4  jammy       Running
+```
+*Note: this tutorial uses [self-signed certificates](https://en.wikipedia.org/wiki/Self-signed_certificate); self-signed certificates should not be used in a production cluster.*
+
+To enable TLS on Charmed PostgreSQL, relate the two applications:
+```shell
+juju relate postgresql tls-certificates-operator
+```
+
+### Add external TLS certificate
+Use `openssl` to connect to the PostgreSQL and check the TLS certificate in use:
+```shell
+> openssl s_client -starttls postgres -connect 10.89.49.129:5432 | grep Issuer
+...
+depth=1 C = US, CN = Tutorial CA
+verify error:num=19:self-signed certificate in certificate chain
+...
+```
+Congratulations! PostgreSQL is now using TLS certificate generated by the external application `tls-certificates-operator`.
+
+
+### Remove external TLS certificate
+To remove the external TLS and return to the locally generate one, unrelate applications:
```

Comment:
Wouldn't be better to only tell the user that no certificate is being used instead of mentioning a local generated certificate?

Reply:
Oups, it was a copy&paste miss from MySQL.
There is no local TLS for PostgreSQL. Removed. Tnx!